### PR TITLE
Added cin_cam reading, as well as POINT3D

### PR DESCRIPTION
**Pull request by OndrikB**
I don't know if there's any errors. If you find any, tell me.

### CODE REVIEW

File: SAGESharp/SLB/Cinematic/Camera/Frame.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SAGESharp.SLB.Cinematic.Camera
+{
+    public sealed class Frame : IEquatable<Frame>
+    {
+
+        internal const int BINARY_SIZE = 28;
```
Comment (Vraiment) [inline SAGESharp/SLB/Cinematic/Camera/Frame.cs]: Not needed

File: SAGESharp/SLB/Cinematic/Camera/Camera.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+using System.Text;
+
+namespace SAGESharp.SLB.Cinematic.Camera
+{
+    public sealed class Camera : IEquatable<Camera>
+    {
+
+
+
+        internal const int BINARY_SIZE = 20; //TODO: I don't know what I should count
```
Comment (Vraiment) [inline SAGESharp/SLB/Cinematic/Camera/Camera.cs]: Not needed

File: SAGESharp/SLB/Cinematic/Camera/Frame.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            {
+                return false;
+            }
+
+            return Time == other.Time &&
+                Position == other.Position &&
+                Target == other.Target;
+        }
+
+        public override string ToString()
```
Comment (Vraiment) [inline SAGESharp/SLB/Cinematic/Camera/Frame.cs:29]: Use string interpolation:

`return $"Time={Time}, Position={Position}, Target={Target}"`

File: SAGESharp/SLB/Point3D.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            if (other == null)
+            {
+                return false;
+            }
+
+            return X == other.X &&
+                Y == other.Y &&
+                Z == other.Z;
+        }
+        public override string ToString()
```
Comment (Vraiment) [inline SAGESharp/SLB/Point3D.cs]: Use string interpolation:

`return $"X={X}, Y={Y}, Z={Z}";`

File: SAGESharp/SLB/Cinematic/Camera/Camera.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            }
+
+            return ViewAngle == other.ViewAngle &&
+                SpinMaskTimes1 == other.SpinMaskTimes1 &&
+                SpinMaskTimes2 == other.SpinMaskTimes2 &&
+                SpinMaskTimes3 == other.SpinMaskTimes3 &&
+                Frames.SafeSequenceEquals(other.Frames);
+        }
+
+        public override string ToString()
```
Comment (Vraiment) [inline SAGESharp/SLB/Cinematic/Camera/Camera.cs]: Use string interpolation:

```
return $"ViewAngle={ViewAngle}," +
    $"SpinMaskTimes1={SpinMasktimes1}," +
    $"SpinMaskTimes2={SpinMasktimes2}," +
    $"SpinMaskTimes3={SpinMasktimes3}," +
    $"Frames={Frames?.Let(frames => "[(" + string.Join("), (") + ")]") ?? "null"}";
```

You will also need to add: `using Konvenience;` at the top

**Review by Vraiment — CHANGES_REQUESTED** [review]
Also, not as important but can you remove the duplicated empty lines and leave a single empty line between method names (ex: `Point3D.ToString()` doesn't have a preceding empty line)

File: SAGESharp/SLB/Cinematic/Camera/Frame.cs (hunk 3 above)
Comment (Vraiment) [reply SAGESharp/SLB/Cinematic/Camera/Frame.cs:29]: We missed a semicolon at the end of the line :P

File: SAGESharp/SLB/Cinematic/Camera/Camera.cs (hunk 5 above)
Comment (Vraiment) [reply SAGESharp/SLB/Cinematic/Camera/Camera.cs]: I missed here a chunk, the last line should be:

```
    $"Frames={Frames?.Let(frames => "[(" + string.Join("), (", frames) + ")]") ?? "null"}";
```

**Review by Vraiment — CHANGES_REQUESTED** [review]
See comments

**Review by Vraiment — APPROVED** [review]

**Review by epicabsol — APPROVED** [review]

**Comment by Vraiment** [comment]
Tested reading a couple of files with these clases and the smart reader and it seems it works so I'm merging these.

**Pull request merged**